contracts/multisig/src/key.rs

```diff
@@ -72,6 +88,10 @@ impl Signature {
             (Signature::Ecdsa(sig), PublicKey::Ecdsa(pub_key)) => {
                 ecdsa_verify(msg.into(), sig.as_ref(), pub_key.as_ref())
             }
+            (Signature::Ed25519(sig), PublicKey::Ed25519(pub_key)) => {
+                ed25519_verify(msg.into(), sig.as_ref(), pub_key.as_ref())
+            }
+            _ => Err(ContractError::KeyTypeMismatch),
```

Comment:
I don't like that this case now swallows any future additions without breaking (i.e. we might forget to extend it). Since it's ensured that the key type matches for both the key and the signature, the match should be modified to match on the key type, which eliminates any ambiguity and makes the discard case unnecessary.

Reply:
That makes sense, I'll update.